tools/codegen/api/python.py

```diff
@@ -447,7 +447,10 @@ def has_tensor_options(f: NativeFunction) -> bool:
 #
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ #
 
-def argument_type_str(t: Type) -> str:
+# 'simple_type' was introduced by the old codegen, which is slightly
+# different from the python schema type, e.g.: doesn't have '?' suffix
+# for optional Tensor/TensorList; doesn't have '[size]' suffix for list type.
```

Comment:
If `simple_type` is gonna be temporary would be nice to have some explanation of how we can kill it later

Reply:
@ezyang simple_type is in the output file, e.g.:
```
annotated_args = {
    torch._C._VariableFunctions._cast_Byte: [{'name': 'self', 'simple_type': 'Tensor'}],
```

It's referenced in test/test_overrides.py - if it's the only use case then we can fix later. Not sure if there are other uses.

Reply:
The test/test_overrides.py use is definitely incidental. What they needed there was an easy way to tell if an argument would accept overrides or not, and I think they just implemented it via `simple_type` because it was easier that way. But no intrinsic relationship there.